libphal/phal_dump.C

```diff
@@ -169,473 +219,574 @@ void writeSbeData(const std::filesystem::path& dumpPath, uint32_t reasonCode,
  *
- * @throw std::runtime_error Throws if there is an error in collecting the
- * PIBMEM dump data or writing to the file.
+ * @tparam DumpRegVal A templated vector of various possible dump type registers
+ * @param dumpRegs Dump register
+ * @param basePath File base path
+ * @param dumpType Type of dump, e.g; pibms/pibmem/sbe local reg etc
  */
-void collectPIBMEMDump(struct pdbg_target* proc,
-		       const std::filesystem::path& dumpPath,
-		       const std::string& baseFilename)
+template <typename DumpRegVal>
+void writeDumpFileForDumpContents(std::vector<DumpRegVal>& dumpRegs,
+				  std::filesystem::path& basePath,
+				  const std::string& dumpType)
 {
-	// Define these constants and types as per your requirement
-	const uint32_t pibmemDumpStartByte = 0;	      // Starting byte for dump
-	const uint32_t pibmemDumpNumOfByte = 0x7D400; // Number of bytes to read
-	const user_options userOptions =
-	    INTERMEDIATE_TILL_INTERMEDIATE; // User options, define as per your
-					    // code
-	const bool eccEnable = false; // ECC enabled/disabled
-
-	std::vector<array_data_t>
-	    pibmemContents; // Type array_data_t should be defined in your code
-
-	ReturnCode fapiRc =
-	    p10_pibmem_dump(proc, pibmemDumpStartByte, pibmemDumpNumOfByte,
-			    userOptions, pibmemContents, eccEnable);
-	if (fapiRc != FAPI2_RC_SUCCESS) {
+	try {
+		writeDumpFile(reinterpret_cast<char*>(&dumpRegs[0]),
+			      sizeof(DumpRegVal) * dumpRegs.size(), basePath);
+	} catch (const dumpError_t& e) {
 		log(level::ERROR,
-		    "Failed in p10_pibmem_dump for proc=%s, rc=0x%08X",
-		    pdbg_target_path(proc), fapiRc);
-		throw std::runtime_error("p10_pibmem_dump failed");
-	}
-	std::vector<uint64_t> dumpData;
-	for (auto& data : pibmemContents) {
-		dumpData.push_back(data.read_data);
+		    "Error in writing %s "
+		    "file "
+		    "errorMsg=%s",
+		    dumpType, e.what());
+		throw;
 	}
-
-	std::string dumpFilename = baseFilename + "p10_pibmem_dump";
-	std::filesystem::path basePath = dumpPath / dumpFilename;
-
-	// Writing the PIBMEM data to a file
-	writePIBMEMDataToFile(dumpData, basePath);
 }
 
 /**
- * @brief Writes PPE (Programmable Processing Element) state data to a file.
- * @param[in] ppeState A vector of `DumpPPERegValue` structures containing the
- * PPE state data to be written.
- * @param[in] basePath The filesystem path of the file to which the data will be
- * written.
+ * @brief Collects the PPE state data for both p10 and Odyssey
  *
- * @throw std::runtime_error Throws if there is an error opening the file or
- * during the file write operation.
+ * @param failingUnit Position of the proc containing the failed SBE
+ * @param baseFilename Base file name of the dump
+ * @param dumpPath The dump path
+ * @param sbeTypeId The type of failed SBE chip, i.e.; p10 or Odyssey
  */
-void writePPEStateToFile(const std::vector<DumpPPERegValue>& ppeState,
-			 const std::filesystem::path& basePath)
+void collectPPEStateData(struct pdbg_target* target,
```

Comment:
why. the order of functons changed?

Reply:
I tried to declare and define the functions as near as possible of theirs usages and in group for easy readability and standard

Reply:
any such changes with a different pull request, keep the commit to the scope of the change

Reply:
Will be discussing it in a meeting perhaps.